google/cloud/pubsublite/internal/default_routing_policy.h

```diff
@@ -30,10 +29,10 @@ GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
 // Calculates the val^pow % mod while accounting for overflow.
 // Needed because after calculating `big_endian[i]` % `mod` in `GetMod`, we need
 // to account for its position in the array by multiplying it by an offset.
-std::uint64_t ModPow(std::uint64_t val, std::uint64_t pow, std::uint32_t mod);
+std::uint64_t ModPow(std::uint64_t val, std::uint64_t pow, Partition mod);
```

Comment:
nit. for "math" functions like ModPow and GetMod, probably better to just use std::uint32_t, makes it easier to think about overflows.